docs/conditional-gatherer/README.md

```diff
@@ -1,185 +1,137 @@
-                        "alert": "SamplesImagestreamImportFailing",
-                        "annotations": {
-                            "message": "Always firing"
-                        },
-                        "expr": "1 > bool 0",
-                        "for": "1s",
-                        "labels": {
-                            "severity": "warning"
-                        }
-                    }
-                ]
-            }
-        ]
-    }
-}' | oc apply -f -
-```
+### Pod Logs
 
-4. Wait for the alert to fire:
-```bash
-export PROMETHEUS_HOST=$(oc get route -n openshift-monitoring prometheus-k8s -o jsonpath='{@.spec.host}')
-export INSECURE_PROMETHEUS_TOKEN=$(oc get secret $(oc get sa prometheus-k8s -n openshift-monitoring -o json | jq .secrets[0].name | tr --delete \") -n openshift-monitoring -o json | jq .metadata.annotations.\"openshift.io/token-secret.value\" | tr --delete \")
-curl -g -s -k -H 'Cache-Control: no-cache' -H "Authorization: Bearer $INSECURE_PROMETHEUS_TOKEN" "https://$PROMETHEUS_HOST/api/v1/query" --data-urlencode 'query=ALERTS{alertstate="firing",alertname="SamplesImagestreamImportFailing"}' | jq ".data.result[]"
-```
+The `pod_logs` gathering function is a powerful tool to collect logs from specific pods based on various criteria. Here's an example configuration:
 
-The output should be:
 ```json
 {
-  "metric": {
-    "__name__": "ALERTS",
-    "alertname": "SamplesImagestreamImportFailing",
-    "alertstate": "firing",
-    "severity": "critical"
-  },
-  "value": [
-    1652363876.855,
-    "1"
-  ]
+    "pod_logs": {
+        "resource_filter": {
+            "namespaces": ["openshift-monitoring"],
+            "pod_name_regex_filter": "thanos-"
+        },
+        "log_filter": {
+            "messages_to_search": ["connect: connection refused"],
+            "tail_lines": 100,
+            "previous": false
+        }
+    }
 }
 ```
 
-5. Make metrics work by forwarding the endpoint and setting INSECURE_PROMETHEUS_TOKEN environment variable:
-```bash
-export INSECURE_PROMETHEUS_TOKEN=$(oc get secret $(oc get sa prometheus-k8s -n openshift-monitoring -o json | jq .secrets[0].name | tr --delete \") -n openshift-monitoring -o json | jq .metadata.annotations.\"openshift.io/token-secret.value\")
-```
-```bash
-# run this command in a separate terminal
-sudo kubefwd svc -n openshift-monitoring -d openshift-monitoring.svc -l app.kubernetes.io/instance=k8s --kubeconfig $KUBECONFIG
-```
-
-6. Run the operator and wait for an archive containing `conditional/` directory.
+In this example:
 
-7. Restore the backup:
-```bash
-oc apply -f prometheus-rules.back.json
-```
+- `resource_filter.namespaces`: Specifies the namespaces to search for resources.
+- `resource_filter.pod_name_regex_filter`: Uses a regex to filter pod names.
+- `log_filter.messages_to_search`: Lists messages to search for in the logs.
+- `log_filter.tail_lines`: Specifies the number of lines to retrieve from the end of logs.
```

Comment:
How does it work with `messages_to_search`?

Reply:
It doesn't impact the `messages_to_seach`. It only limits the number of lines fetched from the Openshift API per container.

Reply:
> It doesn't impact the messages_to_search.

IIUC, parameters like `tail_lines` specify a portion of the log where lines matching `messages_to_search` will be searched.

Note that this is very different from "IO gathers the last `tail_lines` lines matching `messages_to_search`". The current documentation allows this interpretation.

Reply:
P.S. Also note that Ivan requested to be able to gather a log tail (e.g. for terminating pods). So it could also mean "gather all lines that match messages and a log tail specified by the other parameters". 

I don't understand what your emoji reaction means. Are you planning to take any action based on this feedback?